File: solutions/ocaml/036-rotate-image.ml
(*
  Problem 36: Rotate Image (LeetCode 48)
  Difficulty: Med
  Language: OCaml
*)
let rotate matrix =
  let n = Array.length matrix in
  Array.init n (fun i ->
    Array.init n (fun j -> matrix.(n - 1 - j).(i)))

File: solutions/rust/036-rotate-image.rs
/**
 * Problem 36: Rotate Image (LeetCode 48)
 * Difficulty: Med
 * Language: Rust
 */
pub fn rotate(matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
    let n = matrix.len();
    (0..n).map(|i|
        (0..n).rev().map(|j| matrix[j][i]).collect()
    ).collect()
}
